src/ex3/commands.js

```diff
@@ -0,0 +1,104 @@
+import { writeToFile, readFromFile } from "./fs.js";
+import {
+  checkItem,
+  capitalize,
+  convertToArray,
+  deletebyIndex,
+  deleteByName,
+} from "./utils.js";
+import { itemManager } from "../ex2/ItemManager.js";
+import { success, error, blue } from "./chalk.js";
+
+export async function addItem(item) {
+  let pokemonArray = [];
+
+  const inputArray = item.split(",");
+  inputArray.forEach(async (elm) => {
+    if (isNaN(elm)) {
+      await addToDo(elm);
+    } else {
+      pokemonArray.push(elm);
+    }
+  });
+
+  await addPokemon(pokemonArray);
+
+  async function addToDo(item) {
+    const capitalizedItem = capitalize(item);
+    await writeToFile(capitalizedItem, true, "a+");
+    console.log(success(`\n\nSuccessfully added: ${capitalizedItem}`));
+  }
+
+  async function addPokemon(pokemonArray) {
+    const allPromises = itemManager.createPromises(pokemonArray); // add promises to allPromises array
+    const pokemonData = await Promise.all(allPromises); // fetch all pokemons simulteniously
+
+    const copyPokemonArray = [...pokemonArray];
+
+    pokemonData.forEach((pokemon) => {
+      if (pokemon) {
+        writeToFile(`Catch ${pokemon.name}`, true, "a+");
+        console.log(success(`\n\nSuccessfully added: Catch ${pokemon.name}`));
+
+        const index = copyPokemonArray.findIndex((elm) => elm == pokemon.id);
+        copyPokemonArray.splice(index, 1);
```

Comment:
you've created a inefficient function here since you go over each pokemon in the array and then for each go over the entire array to find the index and then again to splice

Reply:
there are a few ways to do this - first if you don't mind printing the found/ not found mixed you can just print the not found if the pokemon doesn't exist and return adn then preint 'catch...' if ti does. 

However, I assume you wanted to print all the not founds at the end after wards. In this case it would be better to either use a Set so you don't need to search for the index or to just do a forEach that splits the array into two arrays, one of found and one of not found and then print the desired messages for each, that way you have something that's O(n) since it just goes over the full array twice and not O(n^2)